python/examples/schedule/transform_a_payload_according_to_a_schedule.py

```diff
@@ -0,0 +1,120 @@
+            zero_init = linalg.fill(c0, outs=[empty])
+            # CHECK: %[[A_X_WEIGHTS:.*]] = linalg.matmul ins(%[[MATRIX_A]], %[[WEIGHTS]]{{.*}}) outs(%[[ZERO_INIT]]
+            A_x_weights = linalg.matmul(matrixA, weights, outs=[zero_init])
+            empty2 = tensor.empty(matrixType.shape, matrixType.element_type)
+            zero_init2 = linalg.fill(c0, outs=[empty2])
+            # CHECK: %[[RES:.*]] = linalg.matmul ins(%[[MATRIX_B]], %[[WEIGHTS]]{{.*}}) outs(%[[A_X_WEIGHTS]]
+            B_x_weights = linalg.matmul(matrixB, weights, outs=[zero_init2])
+            # CHECK-NOT: linalg.add
+            added = linalg.add(A_x_weights, B_x_weights, outs=[empty])
+            # CHECK: return %[[RES]]
+            return added
+
+    print(payload)
+    return payload
+
+
+def example_schedule() -> Module:
+    """Most basic schedule which doesn't just wrap a pass -- wraps a single rewrite pattern."""
+    print("NOTE: example schedule module:")
+    schedule_module = Module.create()
+    schedule_module.operation.attributes["transform.with_named_sequence"] = (
+        UnitAttr.get()
+    )
+    with InsertionPoint(schedule_module.body):
+        named_seq = transform.named_sequence(
+            "__transform_main",
+            input_types=[transform.any_op_t()],
+            result_types=[],
+            arg_attrs=[{"transform.readonly": UnitAttr.get()}],
+        )
+
+        with InsertionPoint(named_seq.body):
+            func = structured.MatchOp.match_op_names(
+                named_seq.bodyTarget, ["func.func"]
+            )  # TODO: fix syntax upstream
+            with InsertionPoint(transform.apply_patterns(func).patterns):
+                Operation.create(
+                    "transform.apply_patterns.linalg.fold_add_into_dest"
+                )  # TODO: expose dedicated builder upstream
+            transform.yield_([])
+
+    print(schedule_module)
+    return schedule_module
+
+
+with Context(), Location.unknown():
+    payload = example_payload()
+    schedule_module = example_schedule()
+    schedule: transform.NamedSequenceOp = schedule_module.body.operations[0]
+
+    print(
+        "NOTE: result of applying schedule to payload directly within Python process:"
+    )
+    schedule.apply(payload)
+    print(payload)
+
+    # Demonstrate applying a schedule from file to a payload from file
+    with (
+        tempfile.NamedTemporaryFile("w", prefix="payload_") as payload_file,
+        tempfile.NamedTemporaryFile("w", prefix="schedule_") as schedule_file,
+    ):
+        print(payload, file=payload_file, flush=True)
+        print("NOTE: Have dumped payload to temp file:", payload_file.name)
+        print(schedule_module, file=schedule_file, flush=True)
+        print("NOTE: Have dumped schedule to temp file:", schedule_file.name)
+
+        cmdline = [
+            "python",
+            "-m",
+            "lighthouse.schedule",
+            schedule_file.name,
+            payload_file.name,
+        ]
```

Comment:
I'd rather we didn't generate python invocation lines from within Python. This means that there is no separation of concerns and it's quite hard to extract/learn _what_ exactly needs to happen (i.e. what the steps are).

In particular, to me, this script is trying to achieve three things in one go:
1. Generate Payload IR (most likely generating Linalg or Vector Ops)
2. Generate Schedule IR (generates TD Ops).
3. Generate cmdline and invoke it (orthogonal to MLIR generation).

These are 3 separate tasks, each of which comes with its own set of complexities and challenges. Also, ATM, both `__main__.py` and `transform_a_payload_according_to_a_schedule.py` are runnable. So, IIUC, there are two ways to run `transform_a_payload_according_to_a_schedule.py`? If "yes", why?